packages/server/src/notifier/model/notifications.ts

```diff
@@ -17,11 +17,9 @@ export const notificationsFromEvent =
       (acc, { id, receiveEmails }) => ({ ...acc, [id]: receiveEmails }),
       {}
     )
+    const allMemberIds = Object.keys(membersEmailPreferences).map(Number)
     const notifsByMembers = event.potentialNotifications.flatMap<PotentialNotifByMember>(
-      getEventsByMember(
-        subscriptions,
-        allMembers.map(({ id }) => id)
-      )
+      getEventsByMember(subscriptions, allMemberIds)
```

Comment:
The performance issue I mentioned is that the partially applied function (`notificationsFromEvent(subscription)`) is called by the `flatmap` once per events, and on each call it recomputes both `membersEmailPreferences` and `allMemberIds` (which depends on the total number of users in the db).

So I was suggesting to take `membersEmailPreferences` and `allMemberIds` out of the inner function that runs once per events, into the outer function's scope which runs just once. Alternatively you could also compute these in the `createNotifications` function and pass them to `notificationsFromEvent` as additional parameters.

Reply:
Ahh I see now, I had trouble understanding what you suggested